scripts/pipeline/orthology_benchmark.py

```diff
@@ -17,18 +17,134 @@
-    """Docstring"""
+def dump_genomes(species_list: List[str], species_set_name: str, host: str, port: int,
+                 user: str, out_dir: str) -> None:
+    """Dumps canonical peptides of protein-coding genes for a specified list of species.
+
+    Peptides are dumped from the latest available core databases to FASTA files
+    using `dump_gene_set_from_core.pl`.
+
+    Args:
+        species_list: A list of species (genome) names to dump.
+        species_set_name: Species set (collection) name.
+        host: Database host.
+        port: Host port.
+        user: Server username.
+        out_dir: Directory to place `species_set_name/core_name.fasta` dumps.
+
+    Raises:
+        FileExistsError: If directory `out_dir/species_set_name` already exists.
+        sqlalchemy.exc.OperationalError: If `user` cannot read from `host:port`.
+        RuntimeError: If no cores are found for any species in `species_list` (`species_set_name`)
+            on `host:port`.
+
+    """
+    cores = get_core_names(species_list, host, port, user)
+    dump_cores = [core for species, core in cores.items() if core != ""]
+
+    if len(dump_cores) == 0:
+        raise RuntimeError(f"No cores found for the species set '{species_set_name}' on the specified host.")
+
+    dumps_dir = os.path.join(out_dir, species_set_name)
+    os.mkdir(dumps_dir)
+
+    script = os.path.join(os.environ["ENSEMBL_ROOT_DIR"], "ensembl-compara", "scripts", "dumps",
+                          "dump_gene_set_from_core.pl")
+
+    for core in dump_cores:
+        out_file = os.path.join(dumps_dir, f"{core}.fasta")
+        subprocess.run([script, "-core-db", core, "-host", host, "-port", str(port),
+                       "-outfile", out_file], capture_output=True, check=True)
+
+
+def find_latest_core(core_names: List[str]) -> str:
+    """Returns the latest core database among the available ones.
+
+    Args:
+        core_names: A list of cores for a species of interest.
+
+    Raises:
+        RuntimeError: If `core_names` is empty.
+
+    Returns:
+        Name of the latest core database (the latest Ensembl release, the latest version).
+
+    """
+    if len(core_names) == 0:
+        raise RuntimeError("Empty list of core databases. Cannot determine the latest one.")
+
+    rel_ver = [name.split("_core_")[1].split("_") for name in core_names]
+    rel_ver_int = [list(map(int, i)) for i in rel_ver]
+    rel_ver_arr = numpy.array(rel_ver_int)
+
+    n_cols = rel_ver_arr.shape[1]
+    i = 0
+    while i < n_cols:
+        # Find max value in the i-th column
+        # For the next iteration (i+1) consider only rows where i-th column == max value
+        max_col = numpy.amax(rel_ver_arr[:, i])
+        rows_ind = numpy.where(rel_ver_arr[:, i] == max_col)[0]
+        rel_ver_arr = rel_ver_arr[[rows_ind], :][0]
+        i += 1
+
+    latest_rel_ver = '_'.join(map(str, rel_ver_arr[0]))
+    core_name = [core for core in core_names if latest_rel_ver in core][0]
```

Comment:
Although it works, I find this code a little bit complex to follow in one go. This is just a suggestion to simply it and improve its readability:
```suggest
    versions = [name.split('_')[-2:] for name in core_names]
    latest = 0
    for i, row in enumerate(versions[1:], 1):
        if (row[0] > versions[latest][0] or (row[0] == versions[latest][0] and row[1] > versions[latest][1])):
            latest = i
    core_name = core_names[latest]
```
(hope I haven't missed any functionality)

Reply:
I just saw this. Hm hm, I've read it twice and couldn't understand it 👼  Out of curiosity, what would it return in case of core versions `105_279, 52_105_279, 106_3`?

Reply:
My code just cares about the release version (ignoring EG) and core version, so in your list, it would return the `107_2` one.